### ISSUE TITLE
SessionShouldBeLoggedOutTest is not authoratitive

### ISSUE THREAD

**Issue by justinedelson** [Cognifide/AEM-Rules-for-SonarQube#58]
The file used by SessionShouldBeLoggedOutTest has a mix of compliant and non-compliant methods but only checks that at least one method is non-compliant. This can lead to false positives/negatives. It would be better to break these out into separate test files and use parameterized tests in JUnit.

**Comment by chutch** [comment]
Well, actually I think that the test was OK. It works in following manner: if there is a non-compliant line, it expects it to be commented. If it is not marked, it will break the test. On the other hand, if a line is expected to be non-complaint and the rule doesn't find it - it will also break the build. It just expects fixed number of issues in specified lines. Or maybe I'm missing something... it's late. :-)

But.. I do see value in this breakdown when it comes to readability :-) it's quite a leap.

**Comment by justinedelson** [comment]
ah. I wasn't aware that those comments were actually meaningful for the test. That wasn't intuitive to me, but I'm also a noob at this rule writing stuff. Mostly just hacking on what you guys have already done :)

**Comment by toniedzwiedz** [comment]
@justinedelson I can see how that can be counterintuitive. @chutch is this something project-specific or a pattern generally used in the testing of rules for SonarQube? How about adding a comment or maybe a section in the README explaining how these tests work? It would be helpful for those familiar with AEM but not this project in case they want to contribute.

**Comment by chutch** [comment]
Yeah, that can be misleading, it's not obvious. It's a new school for writing these tests and it's becoming a standard. It's far more readable and maintainable than it was before. Here's a snippet of code of how it looked previously for the record:

```
Set<CheckMessage> checkMessages = file.getCheckMessages();
		checkMessagesVerifier.verify(checkMessages)
				.next().atLine(31).withMessage("ResourceResolver should be closed in finally block.")
				.next().atLine(82);
```

Anyways, now each test case is much easier to understand, so.. keeping it. @justinedelson feel welcome to hack more or leave your ideas for new rules as GitHub issues :-)
